android/src/main/java/com/urbanairship/reactnative/ReactAirshipPreferences.java

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2017 Urban Airship and Contributors */
+
+package com.urbanairship.reactnative;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.urbanairship.UAirship;
+import com.urbanairship.reactnative.events.NotificationOptInEvent;
+
+/**
+ * Stores shared preferences and checks preference-dependent state.
+ */
+public class ReactAirshipPreferences {
+
+    private static ReactAirshipPreferences sharedInstance = new ReactAirshipPreferences();
+
+    private SharedPreferences preferences;
+
+    private static final String SHARED_PREFERENCES_FILE = "com.urbanairship.reactnative";
+
+    protected static final String NOTIFICATIONS_OPT_IN_KEY = "NOTIFICATIONS_OPT_IN_KEY";
+
+    /**
+     * Returns the shared {@link ReactAirshipPreferences} instance.
+     *
+     * @return The shared {@link ReactAirshipPreferences} instance.
+     */
+    static ReactAirshipPreferences shared() {
+        return sharedInstance;
+    }
+
+    public SharedPreferences getPreferences (Context context) {
```

Comment:
I can just remove it - I added it to expose the shared prefs file to the rest of the module, probably should have been protected